README.md

```diff
@@ -21,233 +24,132 @@ repositories {
 }
 
 dependencies {
-
-    // UniversalAdapter with changes detector with RxJava
-    implementation 'com.github.jacek-marchwicki.recyclerview-changes-detector:universal-adapter-rx:<look-on-release-tab>'
-
-    // UniversalAdapter with changes detector without RxJava
     implementation 'com.github.jacek-marchwicki.recyclerview-changes-detector:universal-adapter:<look-on-release-tab>'
-
-    // Changes Detector and Adapter items (without Android dependencies)
-    implementation 'com.github.jacek-marchwicki.recyclerview-changes-detector:universal-adapter-java:<look-on-release-tab>'
-
-    // Changes Detector (without Android dependencies)
-    implementation 'com.github.jacek-marchwicki.recyclerview-changes-detector:changes-detector:<look-on-release-tab>'
+    // RX support
+    implementation 'com.github.jacek-marchwicki.recyclerview-changes-detector:universal-adapter-rx:<look-on-release-tab>'
 }
 ```
 
 ## How to use
 
-### With Universal Adapter
-
-Implement some Pojo with your data:
+Let's assume that your list consits of a header, songs and footer.
 
-```java
-private class Data implements BaseAdapterItem {
-
-    private final long id;
-    @Nonnull
-    private final String name;
-    private final int color;
-
-    Data(long id, @Nonnull String name, int color) {
-        this.id = id;
-        this.name = name;
-        this.color = color;
-    }
-
-    @Override
-    public long adapterId() {
-        return id;
-    }
-
-    /**
-     * Return true if id matches
-     */
-    @Override
-    public boolean matches(@Nonnull BaseAdapterItem item) {
-        return item instanceof Data && (((Data) item).id == id);
-    }
+- Implement data models for all list elements
 
-    /**
-     * Return true if items are equal
-     */
-    @Override
-    public boolean same(@Nonnull BaseAdapterItem item) {
-        return equals(item);
-    }
+```kotlin
+data class HeaderItem(val text: String, val songsCount: Int, override val itemId: Any = text) : DefaultAdapterItem()
+data class SongItem(val id: String, val title: String, val imageUrl: String, override val itemId: Any = id, val onSongClick: (id: String) -> Unit) : DefaultAdapterItem()
+data class Footer(override val itemId: Any = NO_ID) : DefaultAdapterItem()
```

Comment:
suggestion: it would be better if you could add line breaks. Right now it's hard to read this.